File: regular_expressions/pl/ulists.pl
#!/usr/bin/perl

##############################################
## using PL regular expressions to find and
## unnumbered list items.
## @author: vladimir kulyukin
##############################################

use strict;
use warnings;

my $UL_TXT_01 = '
<ul>
<li>John   Balbiro	A1001	john.balbiro@usu.edu</li>
<li>Alice  Nelson	A0011	alice.nelson@workflow.net</li>
<li>Jacob  Roberts	A1100	j.s.roberts@gmail.com</li>
</ul>
';

my $ulist_pat = '(<li>(\w|\s|\d|\W)*</li>)';
my @li_items = ();
while ( $UL_TXT_01 =~ /$ulist_pat/g ) {
    push(@li_items, $1);
}
print "@li_items\n";

File: regular_expressions/py/ulists.py
#!/usr/bin/python

######################################################
## using Python regular expressions to find and
## unnumbered list items.
## @author: vladimir kulyukin
######################################################

import re

UL_TXT_01 = '\n\
<ul>\n\
<li>John   Balbiro	A1001	john.balbiro@usu.edu</li>\n\
<li>Alice  Nelson	A0011	alice.nelson@workflow.net</li>\n\
<li>Jacob  Roberts	A1100	j.s.roberts@gmail.com</li>\n\
</ul>\n\
';

ulist_pat = r'(<li>(\w|\s|\d|\W)*</li>)'
li_items = []
for line in re.split(r'\n', UL_TXT_01):
    match = re.match(ulist_pat, line)
    if ( match != None ): li_items.append(match.group(1))
print li_items

